android/src/main/java/com/stripeterminalreactnative/StripeTerminalReactNativeModule.kt

```diff
@@ -652,8 +653,14 @@ class StripeTerminalReactNativeModule(reactContext: ReactApplicationContext) :
     @Suppress("unused")
     fun collectRefundPaymentMethod(params: ReadableMap, promise: Promise) =
         withExceptionResolver(promise) {
-            val chargeId = requireParam(params.getString("chargeId")) {
-                "You must provide a chargeId"
+            val chargeId = params.getString("chargeId")
+            val paymentIntentId = params.getString("paymentIntentId")
+
+            if (chargeId.isNullOrBlank() && paymentIntentId.isNullOrBlank()) {
+                throw TerminalException(
+                    TerminalException.TerminalErrorCode.INVALID_REQUIRED_PARAMETER,
+                    "You must provide a refund Id."
+                )
```

Comment:
Can we also throw an error when both charge ID and PaymentIntent ids are set, we should only accept "one of", seems like the case we're using the chargeID here when a user provides both